functional-samples/tutorial.open-api-reference/sw-tips.js

```diff
@@ -0,0 +1,28 @@
+console.log('sw-tips.js');
+
+// Fetch tip & save in storage
+const updateTip = async () => {
+  const response = await fetch('https://extension-tips.glitch.me/tips.json');
+  const tips = await response.json();
+  const index = Math.floor(Math.random() * tips.length);
+  await chrome.storage.local.set({ tip: tips[index] });
+};
+
+// Create a daily alarm and retrieves the first tip when extension is installed.
+chrome.runtime.onInstalled.addListener(({ reason }) => {
+  if (reason === 'install') {
+    chrome.alarms.create({ delayInMinutes: 1, periodInMinutes: 1440 });
+    updateTip();
+  }
+});
+
+// Retrieve tip of the day
+chrome.alarms.onAlarm.addListener(updateTip);
+
+// Send tip to content script via messaging
+chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
+  if (message.greeting === 'tip') {
+    chrome.storage.local.get('tip').then(sendResponse);
```

Comment:
Please use async await

Reply:
I used `.then()` here b/c the return value of the `onMessage` listener determines if Chrome expects `sendResponse` to be called.

If we use async/await, this function will return a Promise, and Chrome will always expect `sendResponse` to be called.

We could wrap the call to `.get()` in an async IIFE, but it seems like overkill for this tutorial😊

Reply:
I didn't know that, have you got a link to where this is explained. Curious to learn how this works.

Reply:
Thanks ! That's not intuitive. 

One thing: maybe rename the message field from `greeting` to `type` as it better expresses what this field is used for.